filter/addfield/filteraddfield.go

```diff
@@ -40,8 +41,10 @@ func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeFilterC
 
 // Event the main filter event
 func (f *FilterConfig) Event(ctx context.Context, event logevent.LogEvent) (logevent.LogEvent, bool) {
-	if _, ok := event.Extra[f.Key]; ok {
-		return event, false
+	if f.Overwrite == false {
+		if _, ok := event.Extra[f.Key]; ok {
```

Comment:
`if _, ok := event.Extra[f.Key]; ok && !f.Overwrite {`

Reply:
I have incorporated the change.